File: code_samples/csharp/oop_encapsulation.cs
using System;

/// <summary>
/// Oop: Encapsulation
/// AI/ML Training Sample
/// </summary>
public class Encapsulation
{
    private string data;
    
    public Encapsulation()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Encapsulation();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/haskell/oop_encapsulation.hs
-- Oop: Encapsulation
-- AI/ML Training Sample

module Encapsulation where

data Encapsulation = Encapsulation {
    getData :: String
} deriving (Show, Eq)

process :: Encapsulation -> String -> Encapsulation
process obj input = obj { getData = input }

validate :: Encapsulation -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Encapsulation { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
